gadopt/equations.py

```diff
@@ -71,7 +73,7 @@ def residual(self, test, trial, trial_lagged=None, fields=None, bcs=None):
 
 class BaseTerm(ABC):
     """A term in an equation, that can produce the UFL expression for its contribution to the FEM residual."""
-    def __init__(self, test_space, trial_space, dx, ds, dS, **kwargs):
+    def __init__(self, test_space, trial_space, dx, ds, dS, su_advection, **kwargs):
```

Comment:
Similarly, this pollutes every term to require an su_advection argument which I don't think is right

Reply:
as above changed